### PR TITLE
Make `Body` for new routes generic

### LINKED ISSUE #2401: POST Body is tied to HanEthereumEther-HALightLightningBitcoin
## Problem

Currently the body of the POST request used by the REST API to post a halight swap is modelled with a struct (`Body`) that is tied to the protocol pair.

https://github.com/comit-network/comit-rs/blob/d6e1d3499941f69fbf39e4f0d658896f98d4ae22/cnd/src/http_api/routes/index.rs#L102-L108

## Goal

Be able to deserialize all incoming swaps on the REST API without code duplication.

## Recommendation

There are currently one handler function per lightning route. All these function should use the same generic body struct (with different type parameters). As of today, only han-ethereum-ether-halight-lightning-bitcoin route should call `ComitLn` (which would still be specialised to han-eth-eth-halight-ln-btc.

Note: Make sure to keep the deserialisation in `serde_json::Value` and deserialize the body using a reference of the result.

### PR DESCRIPTION
**Pull request by D4nte**
Resolves #2401

### CODE REVIEW

File: api_tests/tests/lightning_routes.ts [hunk 1]
```diff
@@ -78,4 +78,23 @@ describe("Lightning routes tests", () => {
...
         })
     );
+
+    it(
+        "create-herc20-ethereum-erc20-halight-lightning-bitcoin-returns-invalid-body",
+        twoActorTest(async ({ alice, bob }) => {
+            await expect(
+                alice.cnd.createHerc20EthereumErc20HalightLightningBitcoin(
+                    // @ts-ignore
+                    {}
```
[the remark this answers is not in the document]
Comment (D4nte) [reply api_tests/tests/lightning_routes.ts:88]: `any` is incorrect but I got the gist :)

File: cnd/src/swap_protocols/facade2.rs [hunk 2]
```diff
@@ -11,7 +11,7 @@ use std::sync::Arc;
 /// before communication with the other node has started
 #[derive(Clone, Digest, Debug)]
 #[digest(hash = "SwapDigest")]
-pub struct CreateSwapParams {
+pub struct HanEtherHalightCreateSwapParams {
```
[the remark this answers is not in the document]
Comment (D4nte) [reply cnd/src/swap_protocols/facade2.rs:14]: We need to find a better naming pattern, not sure it's possible tho.

**Comment by D4nte** [comment]
> I find the naming scheme is a bit inconsistent but that is something we can fix later :)

Should be fixed now.

**Comment by D4nte** [comment]
bors retry

**Comment by D4nte** [comment]
bors retry

**Comment by thomaseizinger** [comment]
@D4nte 

> memory allocation of 603979776 bytes failed

What are we doing that rustc wants to allocate 600MB and can't do it.
Seems like CircleCI literally does not give us enough RAM to compile cnd ...

**Comment by D4nte** [comment]
> @D4nte
> 
> > memory allocation of 603979776 bytes failed
> 
> What are we doing that rustc wants to allocate 600MB and can't do it.
> Seems like CircleCI literally does not give us enough RAM to compile cnd ...

Thanks dude! I was seeing the `kill -9` in the non-verbose logs and started to think that circle ci must be killing the process for some reason.

**Comment by D4nte** [comment]
> @D4nte
> 
> > memory allocation of 603979776 bytes failed
> 
> What are we doing that rustc wants to allocate 600MB and can't do it.
> Seems like CircleCI literally does not give us enough RAM to compile cnd ...

Ok I am playing around. I looked at few reddit/stackoverflow threads and this resume the situation:

> TLDR; most people just advised me to switch away from circle-ci. I ended up changing my build script to run cargo test -j1 --doc and cargo test -j1 --lib to separate out my two test runs into different processes, and it seemed to help a bit, but that's not really a solution

@thomaseizinger do we have an issue to move the build to not use circle ci anymore?

**Comment by thomaseizinger** [comment]
> > @D4nte
> > > memory allocation of 603979776 bytes failed
> > 
> > 
> > What are we doing that rustc wants to allocate 600MB and can't do it.
> > Seems like CircleCI literally does not give us enough RAM to compile cnd ...
> 
> Ok I am playing around. I looked at few reddit/stackoverflow threads and this resume the situation:
> 
> > TLDR; most people just advised me to switch away from circle-ci. I ended up changing my build script to run cargo test -j1 --doc and cargo test -j1 --lib to separate out my two test runs into different processes, and it seemed to help a bit, but that's not really a solution
> 
> @thomaseizinger do we have an issue to move the build to not use circle ci anymore?

I think we can move to GitHub actions completely, I looks very stable to me :)

**Comment by bonomat** [comment]
@thomaseizinger : 

> I think we can move to GitHub actions completely, I looks very stable to me :)
> @bonomat what do you think?

I'd be happy with moving to GitHub actions completely. Only the Mac build seems to be unstable, but we can mark this one as not-required.
